### PR TITLE
Migrating Terraform code to the main repo

### PR DESCRIPTION
**Pull request by derekadombek**
# Pull Request
#201 

This is moving the existing Terraform code to the repo from my local. I started working on this a while back when we weren't sure on direction just yet.

## Description
[Provide a brief description of the changes implemented in this pull request.]

## Related Issues
[Link any related issues or tasks from your project management system.]

## Additional Notes
[Add any additional context or notes that reviewers should know about.]

## Checklist
Please review and complete the following checklist before submitting your pull request:

- [x] I have ensured that the pull request is of a manageable size, allowing it to be reviewed within a single session.
- [x] I have reviewed my changes to ensure they are clear, concise, and well-documented.
- [x] I have updated the documentation, if applicable.
- [x] I have added or updated test cases to cover my changes, if applicable.
- [x] I have minimized the number of reviewers to include only those essential for the review.
- [x] I have notified teammates in the review thread to build awareness.

## Checklist for Reviewers
Please review and complete the following checklist during the review process:

- [ ] The code follows best practices and conventions.
- [ ] The changes implement the desired functionality or fix the reported issue.
- [ ] The tests cover the new changes and pass successfully.
- [ ] Any potential edge cases or error scenarios have been considered.

### CODE REVIEW

File: ops/terraform/envs/dev/data.tf [hunk 1]
```diff
@@ -0,0 +1,3 @@
+data "azurerm_resource_group" "dev" {
+  name = "reportvision-${local.environment}-rg"
```
Comment (derekadombek) [inline ops/terraform/envs/dev/data.tf:2]: @zdeveloper heres the name change for the dev resource group and I also name change all the other resources.

File: .github/workflows/build-deploy-frontend.yml [hunk 2]
```diff
@@ -66,7 +66,7 @@ jobs:
       - name: Upload to Azure blob storage
         shell: bash
         run: |
-          az storage blob upload-batch --account-name reportvisionfrontenddev -d '$web' -s client-build/
+          az storage blob upload-batch --account-name reportvisionfrontenddev -d '$web' -s client-build/ --overwrite
```
Comment (derekadombek) [inline .github/workflows/build-deploy-frontend.yml:69]: bit of scope creep here but missed this little detail on my last pr. this allows us to overwrite all the blob files every time we deploy, making sure deployments change everything from scratch. currently, if theres now id change it will on individual files it will skip it.

**Review by zdeveloper — APPROVED** [review]
LGTM

**Review by zdeveloper — APPROVED** [review]
